test/functional/test_framework/blocktools.py

```diff
@@ -138,6 +138,8 @@ def create_coinbase(height, pubkey=None, extra_output_script=None, fees=0, nValu
         coinbaseoutput.nValue += fees
     if pubkey is not None:
         coinbaseoutput.scriptPubKey = key_to_p2pk_script(pubkey)
+    elif script_pubkey is not None:
+        coinbaseoutput.scriptPubKey = script_pubkey
```

Comment:
In 71d9a7c0:

atm is just a non-blocking nit:

it's weird to add this new ´script_pubkey´ arg that is only set if the ´pubkey´ arg is not provided and not stated anywhere.

Ideally, should unify both into a single arg. So there is no confusion over which of them take precedence over the other.